--- lapack/make.inc.example
+++ lapack/make.inc.example
@@ -16,12 +16,12 @@
 #  and handle these quantities appropriately. As a consequence, one 
 #  should not compile LAPACK with flags such as -ffpe-trap=overflow.
 #
-FORTRAN  = gfortran 
-OPTS     = -O2 -frecursive
+#FORTRAN  = gfortran 
+#OPTS     = -O2 -frecursive
 DRVOPTS  = $(OPTS)
-NOOPT    = -O0 -frecursive
-LOADER   = gfortran
-LOADOPTS =
+NOOPT    = $(OPTS) -O0
+LOADER   = $(FORTRAN)
+LOADOPTS = $(OPTS)
 #
 #  Comment out the following line to include deprecated routines to the
 #  LAPACK library.
@@ -51,15 +51,15 @@
 #
 # CC is the C compiler, normally invoked with options CFLAGS.
 #
-CC = gcc
-CFLAGS = -O3
+#CC = gcc
+#CFLAGS = -O3
 #
 #  The archiver and the flag(s) to use when building archive (library)
 #  If you system has no ranlib, set RANLIB = echo.
 #
-ARCH     = ar
+#ARCH     = ar
 ARCHFLAGS= cr
-RANLIB   = ranlib
+#RANLIB   = ranlib
 #
 #  Location of the extended-precision BLAS (XBLAS) Fortran library
 #  used for building and testing extended-precision routines.  The
@@ -79,3 +79,5 @@
 LAPACKLIB    = liblapack.a
 TMGLIB       = libtmglib.a
 LAPACKELIB   = liblapacke.a
+
+# Settings from qml:
