--- lapack/make.inc.example
+++ lapack/make.inc.example
@@ -16,12 +16,12 @@
 #  and handle these quantities appropriately. As a consequence, one 
 #  should not compile LAPACK with flags such as -ffpe-trap=overflow.
 #
-FORTRAN  = gfortran 
-OPTS     = -O2 -frecursive
+#FORTRAN  = gfortran 
+#OPTS     = -O2 -frecursive
 DRVOPTS  = $(OPTS)
-NOOPT    = -O0 -frecursive
-LOADER   = gfortran
-LOADOPTS =
+NOOPT    = $(OPTS) -O0
+LOADER   = $(FORTRAN)
+LOADOPTS = $(OPTS)
 #
 #  Comment out the following line to include deprecated routines to the
 #  LAPACK library.
@@ -51,15 +51,15 @@
 #
 # CC is the C compiler, normally invoked with options CFLAGS.
 #
-CC = gcc
-CFLAGS = -O3
+#CC = gcc
+#CFLAGS = -O3
 #
 #  The archiver and the flag(s) to use when building archive (library)
 #  If you system has no ranlib, set RANLIB = echo.
 #
-ARCH     = ar
+#ARCH     = ar
 ARCHFLAGS= cr
-RANLIB   = ranlib
+#RANLIB   = ranlib
 #
 #  Location of the extended-precision BLAS (XBLAS) Fortran library
 #  used for building and testing extended-precision routines.  The
@@ -79,3 +79,5 @@
 LAPACKLIB    = liblapack.a
 TMGLIB       = libtmglib.a
 LAPACKELIB   = liblapacke.a
+
+# Settings from qml:
--- lapack/Makefile
+++ lapack/Makefile
@@ -18,13 +18,13 @@
 	./testsecond; ./testdsecnd; ./testieee; ./testversion )
 
 blaslib:
-	( cd BLAS/SRC; $(MAKE) )
+	( cd BLAS/SRC; $(MAKE) double )
 
 cblaslib:
 	( cd CBLAS/src; $(MAKE) )
 
 lapacklib:	lapack_install
-	( cd SRC; $(MAKE) )
+	( cd SRC; $(MAKE) double )
 
 lapackelib: lapacklib
 	( cd LAPACKE; $(MAKE) )
@@ -39,11 +39,11 @@
 	( cd SRC/VARIANTS ; $(MAKE))
 
 tmglib:
-	( cd TESTING/MATGEN; $(MAKE) )
+	( cd TESTING/MATGEN; $(MAKE) double )
 
 lapack_testing:	lib
-	( cd TESTING ; $(MAKE) )
-	./lapack_testing.py
+	( cd TESTING ; $(MAKE) double )
+	if python -c 'import sys;assert sys.version_info[0]==2'; then ./lapack_testing.py; else echo "WARNING: need Python 2 to summarize tests"; fi
 
 variants_testing: lib variants
 	( cd TESTING ; rm -f xlintst* ; $(MAKE)  VARLIB='SRC/VARIANTS/LIB/cholrl.a' ; \
@@ -60,21 +60,12 @@
 	mv stest.out stest_qrll.out ; mv dtest.out dtest_qrll.out ; mv ctest.out ctest_qrll.out ; mv ztest.out ztest_qrll.out )
 	
 blas_testing:
-	( cd BLAS/TESTING; $(MAKE) -f Makeblat1 )
-	( cd BLAS; ./xblat1s > sblat1.out    ; \
-	           ./xblat1d > dblat1.out    ; \
-	           ./xblat1c > cblat1.out    ; \
-	           ./xblat1z > zblat1.out    ) 
-	( cd BLAS/TESTING; $(MAKE) -f Makeblat2 )
-	( cd BLAS; ./xblat2s < sblat2.in     ; \
-	           ./xblat2d < dblat2.in     ; \
-	           ./xblat2c < cblat2.in     ; \
-	           ./xblat2z < zblat2.in     )
-	( cd BLAS/TESTING; $(MAKE) -f Makeblat3 )
-	( cd BLAS; ./xblat3s < sblat3.in     ; \
-	           ./xblat3d < dblat3.in     ; \
-	           ./xblat3c < cblat3.in     ; \
-	           ./xblat3z < zblat3.in     ) 
+	( cd BLAS/TESTING; $(MAKE) -f Makeblat1 double )
+	( cd BLAS; ./xblat1d > dblat1.out    ) 
+	( cd BLAS/TESTING; $(MAKE) -f Makeblat2 double )
+	( cd BLAS; ./xblat2d < dblat2.in     )
+	( cd BLAS/TESTING; $(MAKE) -f Makeblat3 double )
+	( cd BLAS; ./xblat3d < dblat3.in     ) 
 
 cblas_testing: blaslib
 	( cd CBLAS ; $(MAKE) cblas_testing)
